contracts/contracts/ResolutionRegistry.sol

```diff
@@ -30,10 +32,15 @@ contract ResolutionRegistry is CrossDomainRestrictedCalls {
         bytes32 fillHash,
         uint256 resolutionChainId
     ) external restricted(resolutionChainId, msg.sender) {
+        require(
+            fillers[fillHash] == address(0),
+            "Cannot invalidate resolved fillHashes"
```

Comment:
I'm thinking, would it make sense to unify the two mappings?

```
    // mapping from fillHash to filler
    mapping(bytes32 => address) public fillers;
    // mapping from fillHash to validity flag
    mapping(bytes32 => bool) public invalidFillHashes;
```

Why not have just one `mapping(bytes32 => address)` where 
```
address == 0  => unresolved
address == 1  => invalidated
address == x  => resolved x as the filler
```
?

Reply:
Additionally with the upcoming changes from #645 this will split up things even more because the invalidation happens on a fillHash basis but the filler mapping is on a request hash basis.